### ISSUE TITLE
_SinkTransformerStreamSubscription is not a subtype of type 'StreamSubscription

### ISSUE THREAD

**Issue by zijam** [dart-lang/sdk#26757]
Upgraded my Dart VM version from **1.16.1** to **1.17.1** running on OSX 10.11.5
I get following exception when using _**package**_ connection_pool **_version_** 0.1.0+2 from pub.dartlang.org.
```
Unhandled exception:
ConnectionPoolException: Failed to open connection. Cause: type '_SinkTransformerStreamSubscription<List<int>, MongoReplyMessage>' is not a subtype of type 'StreamSubscription<List<int>>' of 'value' where
  _SinkTransformerStreamSubscription is from dart:async
  MongoReplyMessage is from package:mongo_dart/mongo_dart.dart
  List is from dart:core
  int is from dart:core
  MongoReplyMessage is from package:mongo_dart/mongo_dart.dart
  StreamSubscription is from dart:async
  List is from dart:core
  int is from dart:core

#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:938)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:41)
#2      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#3      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)
```

I don't get any exception, when I downgrade my VM to **1.16.1**.
Has something been changed in **dart:async** package for the new version of VM.

**Comment by floitschG** [comment]
I recently improved the generic typing of the `dart:async` library.

I believe that the error message is correct: the `_SinkTransformerStreamSubscription<List<int>, MongoReplyMessage>` is a `StreamSubscription<MongoReplyMessage>` and not `StreamSubscription<List<int>>`.

This seems to be a bug in the mongo package.
I have created a pull request to update the type: https://github.com/mongo-dart/mongo_dart/pull/99
(Tbh I didn't test it. Maybe you could give it a try, and report to the maintainers of the package, if that fixes your issue).

**Comment by vadimtsushko** [comment]
Florian, thank you very much for PR.
I've just published v0.2.8 with that fix aplied.
